### DISCUSSION TITLE
Setting minimum width for tabs in TabContainer

### DISCUSSION THREAD

**Discussion by neruthes** [godotengine/godot-proposals#9590] [GUI]
I intend to enforce a constant width for all tabs. The current behavior is to shrink (adapting to text content) as much as possible to minimize total width. I could not find a way to make this happen in Godot documentation or online discussions. StyleBox does not have a minimum width feature and that looks legit; it appears that TabContainer should solve this on its own. Let me know if any solution is available now.

**Comment by timothyqiu** [comment]
Minimum width technically still allows content that _really_ needs more space to exceed the limit.

I think a better fit is to add something like a `fixed_tab_width` toggle. When enabled, make every tab the same width (as the widest width tab).

**Reply by neruthes** [reply]
If we are to make things extendable enough to cover much more cases...

- Minimum width: Mandate the smallest allowed width.
- Maximum width: Mandate the largest allowed width.
- Opportunistically shrink: Shrink all tabs to the same smallest possible width, but not smaller than minimum width.
- Opportunistically enlarge: Enlarge all tabs to the same largest possible width, but not larger than respecting maximum width.
- Content overflow behavior: Visible, Hidden, or Ellipses.

**Comment by neruthes** [comment]
I wrote a script [NSTabsMgr](https://github.com/nekostein/libnsgdgui/blob/master/src/NSTabsMgr.gd) for this.

Attach it to a stand-alone Node. Feed references of tab buttons and tab contents as 2 arrays. NSTabsMgr listens to button_up events and hides certain nodes according to which button is pressed.
